JWBestPracticeApps/NativeControls/ViewController.swift

```diff
@@ -0,0 +1,359 @@
+            // the indicator view will stop animating and show the controls view
+            if oldState == .buffering {
+                self.indicatorView.stopAnimating()
+            }
+        }
+    }
+    
+}
+
+
+// MARK: - JWPlayerDelegate implementation
+
+extension ViewController: JWPlayerDelegate {
+    
+    func onTime(_ event: JWEvent & JWTimeEvent) {
+        guard let player = self.player else { return }
+       
+        if player.duration > 0 {
+            // Enable controls used for normal streaming
+            timeSlider.isEnabled = true
+            replayButton.isHidden = false
+            forwardButton.isHidden = false
+            
+            // Calculate progress(float) from position(seconds)
+            let progress = Float(event.position / event.duration)
+            self.timeSlider.value = progress
+            timeLabel.text = timeFormatted(Int(event.position))
+        } else {
+            // Hide controls not used for live streaming
+            replayButton.isHidden = true
+            forwardButton.isHidden = true
+            
+            if player.duration == 0 {
+                // Hide slider for live streming
+                timeSlider.isHidden = true
+                timeLabel.text = LIVEString
+            } else {
+                // Show slider
+                timeSlider.isHidden = false
+                
+                // Calculate progress(float) from position(seconds)
+                let absPosition = abs(event.position)
+                let absDuration = abs(player.duration)
+                let progress = Float(absPosition / absDuration)
+                self.timeSlider.value = progress
+                timeLabel.text = String.init(format: "%@-%@", LIVEString, timeFormatted(Int(absPosition)))
+            }
+        }
+    }
+    
+    func onFullscreen(_ event: JWEvent & JWFullscreenEvent) {
+        if event.fullscreen {
+            // Check for the current key window view in fullscreen mode
+            if let fullscreenView = UIApplication.shared.keyWindow {
+                // Remove the controls view from its current superview
+                // and add it to the new fullscreen window view
+                controlsView.removeFromSuperview()
+                fullscreenView.addSubview(controlsView)
+                controlsView.constraintToFlexibleBottom()
+            }
+        } else {
+            // Once the fullscreen mode is deactivated
+            // the controls view should be removed from the fullscreen window view
+            controlsView.removeFromSuperview()
+            containerView.addSubview(controlsView)
+            controlsView.constraintToFlexibleBottom()
+        }
+    }
+    
+}
+
+
+// MARK: - UITableViewDelegate & DataSource implementation
+// This is a brief implementation of TableViewController to show the video list
+
+extension ViewController: UITableViewDelegate, UITableViewDataSource {
```

Comment:
Since table view is not used here, you can remove the table view delegates

Reply:
Seems odd. It was removed in this [commit](https://github.com/jwplayer/jwplayer-ios-bestPracticeApps/pull/20/commits/b7d3a5b55ed2da20dc93d5de0a9c0930d3f9041f)